font/src/ft/mod.rs

```diff
@@ -529,6 +581,84 @@ impl FreeTypeRasterizer {
     }
 }
 
+fn downsample_bitmap(mut bitmap_glyph: RasterizedGlyph, fixup_factor: f64) -> RasterizedGlyph {
+    // Don't try to upscale
+    if fixup_factor >= 1.0 {
+        return bitmap_glyph;
+    }
+
+    let bitmap_buffer = if let BitmapBuffer::RGBA(bitmap_buffer) = bitmap_glyph.buf {
+        bitmap_buffer
+    } else {
+        return bitmap_glyph;
+    };
+
+    let bitmap_width = bitmap_glyph.width as usize;
+    let bitmap_height = bitmap_glyph.height as usize;
+
+    let width = (bitmap_width as f64 * fixup_factor) as usize;
+    let height = (bitmap_height as f64 * fixup_factor) as usize;
+
+    let advance_step = (1.0 / fixup_factor).ceil() as usize;
```

Comment:
I'm not sure if I love `advance_step` as a variable name. It's not really bad, but I feel like it might not be completely obvious.

Ideally this variable name would encompass that this is the height and width of the source grid. Do you have any ideas?